synapseclient/client.py

```
@@ -394,6 +394,16 @@ def _get_client_authenticated_s3_profile(self, endpoint, bucket):
         config_section = endpoint + "/" + bucket
         return self._get_config_section_dict(config_section).get("profile_name", "default")
 
+    def _get_transfer_config_max_threads(self):
+        # note RawConfigParser lowercases so maxThreads -> maxthreads
+        max_threads = self._get_config_section_dict('transfer').get('maxthreads')
+        if max_threads:
+            try:
+                return int(max_threads)
+            except ValueError:
+                self.logger.warning("Invalid transfer.maxThreads config setting (%s), igoring.", max_threads)
```

It would be better to raise an intelligible error so that users can fix their config instead of it failing silently since, by default, no config entry is set for maxThreads

---

Agreed, I think if a user is being intentional about setting the maxThreads, an error should be thrown it it is set incorrectly.